api/v1beta1/nutanixcluster_types.go

```diff
@@ -50,6 +50,14 @@ type NutanixClusterSpec struct {
 	// proxy spec.noProxy list.
 	// +optional
 	PrismCentral *credentialTypes.NutanixPrismEndpoint `json:"prismCentral"`
+
+	// failureDomains configures failure domains information for the Nutanix platform.
+	// When set, the failure domains defined here may be used to spread Machines across
+	// prism element clusters to improve fault tolerance of the cluster.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
```

Comment:
if this is also optional then how do we guarantee that user provides cluster details

Reply:
We should handle this with validation webhook.